stdr_server/src/stdr_server.cpp

```diff
@@ -159,11 +163,19 @@ namespace stdr_server {
     
     //!< Publish the new RFID tag list
     stdr_msgs::RfidTagVector rfidTagList;
+    visualization_msgs::MarkerArray RFIDMarkerArray;
+    
     for(RfidTagMapIt it = _rfidTagMap.begin() ; it != _rfidTagMap.end() ; it++)
     {
       rfidTagList.rfid_tags.push_back(it->second);
+      RFIDMarkerArray.markers.push_back(toMarker(it->second)); 
     }
     _rfidTagVectorPublisher.publish(rfidTagList);
+    if (_sourceVectorPublisherRviz.getNumSubscribers() >= 1)
```

Comment:
Although I see the reason why you don't want to publish if no one is listening, this contradicts the use of a latched publisher. Declaring the publisher as latched (which is the right thing to do here) means that whenever a new subscriber "listens" to the markers topic, will get the last message. When this subscriber is the Rviz, means that even if you launch Rviz well after you insert a new source, the sources will get drawn. So I suggest removing the if-statements preventing the publication of marker message for every source (for all types of sources).